### PR TITLE
Add macOS support

### PR DESCRIPTION
**Pull request by deror1869107**
Hi,

I add macOS support and it can build and work on macOS 10.14.6.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,20 +2,33 @@
 ifeq ($(OS),Windows_NT)
     SUFFIX = dll
 else
-    SUFFIX = so
+    OS := $(shell uname -s)
```
Comment (kuba160) [inline Makefile]: I suggest renaming variable `OS` to `UNAME_S`.

File: Makefile [hunk 2]
```diff
@@ -2,20 +2,33 @@
...
+    ifeq ($(OS),Linux)
+        SUFFIX = so
+    else ifeq ($(OS),Darwin)
+        SUFFIX = dylib
+        DEADBEEF_OSX = /Applications/DeaDBeeF.app
+    endif
```
Comment (kuba160) [inline Makefile]: Instead of having 2 ifs, it would be better to do one check for macos and fall back to `.so` suffix.

Comment (kuba160) [reply Makefile]: Also don't forget to rename to `UNAME_S`.

File: Makefile [hunk 3]
```diff
@@ -2,20 +2,33 @@
...
 endif
 
 CC=gcc
 CXX=g++
 STD=gnu99
 CFLAGS+=-fPIC -I /usr/local/include -I discord-rpc/include -Wall
 CXXFLAGS+=-fPIC -I /usr/local/include -Wall
+ifeq ($(OS),Darwin)
```
Comment (kuba160) [inline Makefile]: `OS` -> `UNAME_S`

File: Makefile [hunk 4]
```diff
@@ -2,20 +2,33 @@
...
+    CFLAGS+=-I $(DEADBEEF_OSX)/Contents/Headers
+    CXXFLAGS+=-I $(DEADBEEF_OSX)/Contents/Headers
+endif
 ifeq ($(DEBUG),1)
 CFLAGS +=-g -O0
 CXXFLAGS +=-g -O0
 endif
 
 PREFIX=/usr/local/lib/deadbeef
+ifeq ($(OS),Darwin)
```
Comment (kuba160) [inline Makefile]: `OS` -> `UNAME_S`

**Comment by deror1869107** [comment]
HI,
Thank you for your reviewing and suggestions!
Please review the code again, thanks!

**Pull request merged**